source/extensions/grpc_credentials/aws_iam/config.cc

```diff
@@ -47,7 +47,8 @@ std::shared_ptr<grpc::ChannelCredentials> AwsIamGrpcCredentialsFactory::getChann
         auto credentials_provider = std::make_shared<Common::Aws::DefaultCredentialsProviderChain>(
             api, Common::Aws::Utility::metadataFetcher);
         auto signer = std::make_unique<Common::Aws::SignerImpl>(
-            config.service_name(), getRegion(config), credentials_provider, api.timeSource());
+            config.service_name(), getRegion(config), credentials_provider, api.timeSource(),
+            std::vector<envoy::type::matcher::v3::StringMatcher>{});
```

Comment:
I suggest having a type defined for `std::vector<envoy::type::matcher::v3::StringMatcher>`  perhaps:

```
using AwsSigV4HeaderExclusionVector = std::vector<envoy::type::matcher::v3::StringMatcher>;
```
That way you have a angle bracket less type (AwsSigV4HeaderExclusionVector) you can use throughout the implementation.

You might also have a hardcoded static entry for an empty vector?  This last bit isn't a hard requirement.